In [1]:
import pandas as pd
from datasets import load_from_disk


/Users/yiqinhuang/anaconda3/envs/nlp_env/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
import pandas as pd
import numpy as np
from tqdm.auto import tqdm
data_dir = '../data'
# data_dir = '../../../bloomberg-research/press-releases/data/s_p_500_backlinks'
source_df = pd.read_json(f'{data_dir}/full-source-scored-data.jsonl.gz', lines=True, compression='gzip', nrows=100)

In [3]:
article_d = load_from_disk('../all-coref-resolved')

In [7]:
candidate_df = pd.read_csv('../data/news_articles_covering_the_same_press_releases.csv')
pr_url_set = set(candidate_df['Target URL'].drop_duplicates().values)
filtered_pr_d = article_d.filter(lambda x: x['article_url'] in pr_url_set, num_proc=10)
filtered_pr_df = filtered_pr_d.to_pandas()

In [21]:
print(filtered_pr_df.iloc[0]['sent_lists'])

['The agreement in principle will resolve actual and potential civil claims by the U.S. Department of Justice, the New York and Illinois Attorneys General, the National Credit Union Administration (as conservator for several failed credit unions) and the Federal Home Loan Banks of Chicago and Seattle, relating to the firmas securitization, underwriting and sale of residential mortgage-backed securities from 2005 to 2007.'
 'The agreement in principle will reduce earnings for the fourth quarter of 2015 by approximately $1.5 billion on an after-tax basis.'
 'Under the terms of the agreement in principle, the firm will pay a $2.385 billion civil monetary penalty, make $875 million in cash payments and provide $1.8 billion in consumer relief.'
 'The consumer relief will be in the form of principal forgiveness for underwater homeowners and distressed borrowers; financing for construction, rehabilitation and preservation of affordable housing; and support for debt restructuring, foreclosure 

In [8]:
"".join(filtered_pr_df.iloc[1]['sent_lists'])

"MINNEAPOLIS, MN — General Mills today announced a voluntary national recall of two-, five- and 10-pound bags of its Gold Medal Unbleached and Bleached All Purpose Flour with a “better if used by” date of March 27, 2024, and March 28, 2024.The recall is being issued for the potential presence of Salmonella Infantis, which was discovered during sampling of the five-pound bag product.This recall affects two date codes of Gold Medal Unbleached All Purpose Flour in the five- and ten-pound bags and two date codes of Gold Medal Bleached All Purpose Flour in the two- and five-pound bags.All other types of Gold Medal Flour are not affected by this recall.Consumers are asked to check their pantries and dispose of the product affected by this recall.\n Guidance from the Food and Drug Administration \n(FDA) and the\n Centers for Disease Control \n(CDC) warn that consumers should refrain from consuming any raw products made with flour.Salmonella Infantis is killed by heat through baking, frying, s

In [17]:
count = 0
for i in range(len(filtered_pr_df)):
    if len("".join(filtered_pr_df.iloc[i]['sent_lists'])) <= 100:
        count += 1
        print("STARTT")
        print("".join(filtered_pr_df.iloc[i]['sent_lists']))

print(count)

STARTT
Not Found The requested URL /article_display.cfm was not found on this server.
STARTT
It looks like nothing was found at this location.Maybe try one of the links below or a search?
STARTT
By continuing to browse this site, you agree to this use.
STARTT
Not FoundThe requested page could not be found
STARTT
Not Found The requested URL /article_display.cfm was not found on this server.
STARTT
403 ERROR The request could not be satisfied.
STARTT
If you typed the address, please check that you typed it correctly.
STARTT
 Skip to content 
 Campbell Soup Company
STARTT
 Skip to content 
 Campbell Soup Company
STARTT
© 2007 Pacific Gas and Electric Company.
STARTT
It is possible you typed the address incorrectly, or that the page no longer exists.
STARTT
 Goldman Sachs 
Document Not Found The document you requested could not be found.
STARTT

We're excited to announce that the Lilly Trials blog has a new home on Lilly TrialGuide!
STARTT

Document Not Found The document you requested cou

In [2]:
a = pd.read_csv('pr_summary.csv')

In [4]:
a.iloc[0]['pr_summary']

'Goldman Sachs has reached an agreement in principle to settle claims related to its securitization, underwriting, and sale of residential mortgage-backed securities from 2005 to 2007. The settlement, which is subject to final documentation, will result in a $1.5 billion reduction in earnings for the fourth quarter of 2015. The firm will pay a $2.385 billion civil penalty, make $875 million in cash payments, and provide $1.8 billion in consumer relief, including principal forgiveness for underwater homeowners, financing for affordable housing, and support for debt restructuring and foreclosure prevention programs. The agreement resolves claims by the US Department of Justice, the New York and Illinois Attorneys General, and other regulatory bodies.'

'Goldman Sachs has reached an agreement in principle to settle claims related to its securitization, underwriting, and sale of residential mortgage-backed securities from 2005 to 2007. The settlement, which is subject to final documentation, will result in a $1.5 billion reduction in earnings for the fourth quarter of 2015. The firm will pay a $2.385 billion civil penalty, make $875 million in cash payments, and provide $1.8 billion in consumer relief, including principal forgiveness for underwater homeowners, financing for affordable housing, and support for debt restructuring and foreclosure prevention programs. The agreement resolves claims by the US Department of Justice, the New York and Illinois Attorneys General, and other regulatory bodies.'
